src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -44,57 +48,182 @@ export class PermissionsTableComponent implements OnInit {
+    let entities = permissions[entityType];
+    if (entities) {
+      return entities;
+    } else {
+      throw Error('unknown entity type');
+    }
+  }
+
+  private setPermissionsForResource(resourceType: string) {
+    let parent = this;
+    let getPermissionsFunctions = {
+      'organization': function() {return parent.organizationService.getPermissions();} ,
+      'project': function() {return parent.projectService.getPermissions();},
+      'collection': null, // TODO
+      'view': null // TODO
+    };
+    let getPermissions = getPermissionsFunctions[resourceType];
+    if (getPermissions) {
+      getPermissions().subscribe((permissions: Permissions) => {
+          this.entities = this.getEntities(this.entityType, permissions);
+          this.otherEntities = this.getOtherEntities(this.entityType);
+        });
+    } else {
+      throw Error('unknown resorce type');
     }
   }
 
   private getOtherEntities(entityType: string): string[] {
     switch (entityType) {
       case 'users':
         // TODO get from usersService instead
-        return ['alicak', 'kubedo', 'jkotrady', 'kulexpipiens'];
+        let userEntities = ['alicak', 'kubedo', 'jkotrady', 'kulexpipiens'];
+        for (let entity of this.entities) {
+          let index = userEntities.indexOf(entity['name']);
+          if (index >= 0) {
+            userEntities.splice(index, 1);
+          }
+        }
+        return userEntities;
       case 'groups':
         // TODO get from groupsService instead
-        return ['directors', 'customers'];
+        let groupEntities = ['directors', 'customers'];
+        for (let entity of this.entities) {
+          let index = groupEntities.indexOf(entity['name']);
+          if (index >= 0) {
+            groupEntities.splice(index, 1);
+          }
+        }
+        return groupEntities;
       default:
         throw Error('unknown entity type');
     }
   }
 
-  public onAdd() {
-    // TODO call remote service
+  public onAdd(selectedName: string) {
+    if (this.addedRoles.length === 0 || selectedName === '') {
+      return;
+    }
+    let parent = this;
+    let resources = {
+      'organization': parent.organizationService,
+      'project': parent.projectService,
+      'collection': null, // TODO
+      'view': null // TODO
+    };
+    let resource = resources[this.resourceType];
+
+    let updatePermissionsFunctions = {
+      'users': function(permission: Permission) {return resource.updateUserPermission(permission);} ,
+      'groups': function(permission: Permission) {return resource.updateGroupPermission(permission);}
+    };
+
+    let updatePermissions = updatePermissionsFunctions[this.entityType];
+    if (updatePermissions) {
+      let permission: Permission = {name: selectedName, roles: this.addedRoles};
+      updatePermissions(permission).subscribe();
+      this.entities.push(permission);
+      this.otherEntities.splice(this.otherEntities.indexOf(selectedName), 1);
```

Comment:
What are `otherEntities`? How are they 'other' from `entities`?

Reply:
The `entities` variable stores data about users who already have any role and the `otherEntities` variable stores data about users with no defined variables for a resource.